src/ophys_etl/modules/segmentation/merge/roi_merging.py

```diff
@@ -0,0 +1,378 @@
+                merger_candidates.append(pair)
+
+        shuffler.shuffle(merger_candidates)
+
+        local_t0 = time.time()
+        timeseries_lookup = update_timeseries_lookup(
+                              merger_candidates,
+                              timeseries_lookup,
+                              sub_video_lookup,
+                              filter_fraction=filter_fraction,
+                              n_processors=n_processors)
+
+        logger.info('updated timeseries lookup '
+                    f'in {time.time()-local_t0:.2f} seconds')
+
+        local_t0 = time.time()
+        self_corr_lookup = create_self_corr_lookup(
+                               merger_candidates,
+                               sub_video_lookup,
+                               timeseries_lookup,
+                               filter_fraction,
+                               n_processors)
+
+        new_merger_metrics = get_merger_metric_from_pairs(
+                                   merger_candidates,
+                                   sub_video_lookup,
+                                   timeseries_lookup,
+                                   self_corr_lookup,
+                                   filter_fraction,
+                                   n_processors)
+
+        logger.info('updated self_corr and calculated metrics in '
+                    f'{time.time()-local_t0:.2f} seconds')
+
+        for pair in new_merger_metrics:
+            metric = new_merger_metrics[pair]
+            if metric >= -1.0*corr_acceptance:
+                merger_to_metric[pair] = metric
+
+        potential_mergers = []
+        merger_metrics = []
+        for pair in merger_to_metric:
+            potential_mergers.append(pair)
+            merger_metrics.append(merger_to_metric[pair])
+
+        # order mergers by metric in descending order
+        potential_mergers = np.array(potential_mergers)
+        merger_metrics = np.array(merger_metrics)
+        sorted_indices = np.argsort(-1*merger_metrics)
+        potential_mergers = potential_mergers[sorted_indices]
+
+        recently_merged = set()
+        wait_for_it = set()
+
+        area_lookup = {}
+        for roi_id in roi_lookup:
+            area_lookup[roi_id] = roi_lookup[roi_id].area
+
+        for merger in potential_mergers:
+            roi_id_0 = merger[0]
+            roi_id_1 = merger[1]
+            go_ahead = True
+
+            # if one of the ROIs no longer exists because
+            # it was absorbed into another, skip
+            if roi_id_0 not in valid_roi_id:
+                go_ahead = False
+            if roi_id_1 not in valid_roi_id:
+                go_ahead = False
+
+            # if one of the ROIs has grown by 10% or more since
+            # the start of this iteration, skip
+            if go_ahead:
+                if roi_id_0 in recently_merged:
+                    if roi_lookup[roi_id_0].area > 1.1*area_lookup[roi_id_0]:
```

Comment:
This can be modularized into a function `has_grown_too_large` or something
Then
```
if has_grown_too_large(roi_id_0) or has_grown_too_large(roi_id_1):
    ...
```
on line 268